tests/lib/rules/no-warning-comments.js

```diff
@@ -257,6 +257,136 @@ ruleTester.run("no-warning-comments", rule, {
                     }
                 }
             ]
+        },
+        {
+            code: "// Comment ending with term followed by punctuation TODO!",
+            options: [{ terms: ["todo"], location: "anywhere" }],
+            errors: [
+                {
+                    messageId: "unexpectedComment",
+                    data: {
+                        matchedTerm: "todo",
+                        comment: "Comment ending with term followed by..."
+                    }
+                }
+            ]
+        },
+        {
+            code: "// Comment ending with term including punctuation TODO!",
+            options: [{ terms: ["todo!"], location: "anywhere" }],
+            errors: [
+                {
+                    messageId: "unexpectedComment",
+                    data: {
+                        matchedTerm: "todo!",
+                        comment: "Comment ending with term including..."
+                    }
+                }
+            ]
+        },
+        {
+            code: "// Comment ending with term including punctuation followed by more TODO!!!",
+            options: [{ terms: ["todo!"], location: "anywhere" }],
+            errors: [
+                {
+                    messageId: "unexpectedComment",
+                    data: {
+                        matchedTerm: "todo!",
+                        comment: "Comment ending with term including..."
+                    }
+                }
+            ]
+        },
+        {
+            code: "// !TODO comment starting with term",
+            options: [{ terms: ["todo"], location: "anywhere" }],
```

Comment:
This comment isn't starting with the term?

Reply:
Fixed it.